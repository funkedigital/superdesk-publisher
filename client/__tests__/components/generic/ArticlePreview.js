import React from "react";
import ArticlePreview from "../../../components/generic/ArticlePreview";
import { render } from "@testing-library/react";
import moment from "moment";

jest.mock('moment', () => () => ({ fromNow: () => '2 days ago', format: () => '2020–01–01T12:00:00+00:00' }));

const article = {
  article_statistics: {
    impressions_number: 0,
    internal_click_rate: 0,
    page_views_number: 0
  },
  body:
    "<p>I adore simple pleasures, said Lord Henry. They are the last refuge of the complex. But I don't like scenes, except on the stage. What absurd fellows you are, bot</p>",
  created_at: "2019–08–30T12:34:56+00:00",
  published_at: "2019–08–30T12:34:56+00:00",
  title: "If you let any one have it but me, Basil, I shall never forgive you",
  updated_at: "2019–08–30T12:34:56+00:00"
};

describe("generic/ArticlePreview", () => {
  it("renders correctly", () => {
    const { container } = render(
      <ArticlePreview article={article} close={jest.fn()} />
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
