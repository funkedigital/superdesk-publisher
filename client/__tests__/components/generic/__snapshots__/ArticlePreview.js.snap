// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generic/ArticlePreview renders correctly 1`] = `
<div
  class="sd-preview-panel"
  style="z-index: 1;"
>
  <div
    class="side-panel side-panel--shadow-right"
  >
    <div
      class="side-panel__header side-panel__header--border-bottom"
    >
      <div
        class="side-panel__tools"
      >
        <a
          class="icn-btn"
          flow="top"
          sd-tooltip="Close"
        >
          <i
            class="icon-close-small"
          />
        </a>
      </div>
      <h3
        class="side-panel__heading"
      >
        Item preview
      </h3>
    </div>
    <div
      class="side-panel__content"
    >
      <div
        class="side-panel-collapsible-header active"
      >
        <div
          class="side-panel-collapsible-header__fixed"
        >
          <p
            class="sd-text__date-and-author"
          >
            <time
              title="2019–08–30T12:34:56+00:00"
            >
              2 days ago
            </time>
            
          </p>
        </div>
        <div
          class="side-panel-collapsible-header__collapsible"
        >
          <div
            class="form__row form__row--small-padding"
          >
            <div
              class="flex-grid flex-grid--wrap-items flex-grid--small-2"
            >
              <div
                class="flex-grid__item"
              >
                <label
                  class="form-label form-label--light"
                >
                  Page views
                </label>
                <p>
                  0
                </p>
              </div>
              <div
                class="flex-grid__item"
              >
                <label
                  class="form-label form-label--light"
                >
                  Comments
                </label>
                <p>
                  0
                </p>
              </div>
            </div>
          </div>
        </div>
        <button
          class="side-panel-collapsible-header__toggle active"
        >
          <i
            class="icon-chevron-down-thin"
          />
        </button>
      </div>
      <div
        class="side-panel__content-block"
      >
        <h3
          class="side-panel__content-block-heading"
        >
          If you let any one have it but me, Basil, I shall never forgive you
        </h3>
        <div
          class="side-panel__content-block-text sp-imageMaxWidth-wrapper"
        >
          <p>
            I adore simple pleasures, said Lord Henry. They are the last refuge of the complex. But I don't like scenes, except on the stage. What absurd fellows you are, bot
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
